src/new-components/password-field/password-field.test.tsx

```diff
@@ -260,4 +261,15 @@ describe('PasswordField', () => {
         expect(inputElement).toHaveValue('password value')
         expect(screen.getByTestId('value')).toHaveTextContent('password value')
     })
+
+    describe('a11y', () => {
+        it('renders with no a11y violations', async () => {
+            const { container } = render(
+                <PasswordField data-testid="password-field" id="custom-id" label="New Password" />,
+            )
+            const results = await axe(container)
+
+            expect(results).toHaveNoViolations()
+        })
```

Comment:
I think this could be expanded to also check for a11y violations in password fields that have a `hint` (because this is supposed to add a `aria-describedby` attribute), and maybe test password fields that are disabled.

This feedback here applies to most of the other field components (`SelectField`, `TextField`). I won't repeat the same comment in those components, but whatever you decide to take from this suggestion here should be replicated in those other too.